Adds resources and removes broken links from Outreach page

mburridge opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->
I've updated the docs/getting-started/outreach.md page to add some new links to resources. I've also removed a couple of broken links.

## What?
<!-- In a few words, what is the PR actually doing? -->
To keep the page up to date as no resources after 2021 were included.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a Post or Page. -->
<!-- 2. Insert a Heading Block. -->
<!-- 3. etc. -->

## Screenshots or screencast <!-- if applicable -->

mburridge committed:
Added resources and removed broken links from Outreach page

mburridge committed:
Adds Dave on WP YouTube channel

ryanwelcher reviewed (approved):
LGTM!

annezazu commented on docs/getting-started/outreach.md:
```diff
@@ -76,6 +78,10 @@ Talks given about Gutenberg, including slides and videos as they are available.
 -   [Gutenberg is Coming (Don’t Be Afraid)](https://training.ithemes.com/webinar/gutenberg-is-coming-dont-be-afraid/), iThemes Training (September 2017)
+-   [Converting a Shortcode into a Block](https://learn.wordpress.org/tutorial/converting-a-shortcode-into-a-block/), Jonathoan Bossenger
+-   [Using the create-block Tool](https://learn.wordpress.org/tutorial/using-the-create-block-tool/), Jonathoan Bossenger
+-   [Ryan Welcher Codes](https://www.youtube.com/@RyanWelcherCodes), YouTube Channel
```
I don't think it makes sense to add people's youtube channels here. I'd recommend removing. There are MANY YouTubers otherwise.

mburridge replied:
Are there many who focus on block development to the extent that Ryan does?

ryanwelcher replied:
@annezazu is the concern the optics around who is selected to be on the list?

annezazu replied:
The concern is around a combination of optics and impact on revenue for YouTubers. I think it's okay to link to individual videos or series. It's the same as linking to someone's entire blog rather than being more specific in my mind.

mburridge replied:
>  It's the same as linking to someone's entire blog rather than being more specific in my mind.

That's a pretty good analogy 🤔. I can totally see your point of view. But I can also see the value of providing ready access to the kind of work that Ryan (and others) are doing in their channels. 

I'd love to get other people's thoughts on this so we can weigh up both sides of the argument and get this call right.

mburridge committed:
Adds suggestions by Birgit

juanmaguitar commented:
@mburridge could we make this PR about just removing broken links?
I think the "add resources" part needs more conversation and should be moved to another PR to not block the value of removing broken links.

juanmaguitar commented:
For the Outreach page content, I would create a Google Doc with suggested content for that page so the conversation about that content is more efficient, and I would take this doc to the next Docs Team meeting.
I think community links should be added carefully because of the risk of preferences.
Any new link under .org shouldn't have any problem (that includes Developer Blog).
I think the more consensus about the content of this page the better.

mburridge committed:
Removes links to YouTube channels

mburridge commented:
> @mburridge could we make this PR about just removing broken links? I think the "add resources" part needs more conversation and should be moved to another PR to not block the value of removing broken links.

I've removed the links to YouTube channels. I've kept the links to courses that I added, and of course the broken links have been removed.

I agree that the content of this page needs further discussion and arrival at a consensus.

Mamaduka commented:
Is this still relevant or can we close the PR?
